tests/integration/Util/ContentTypeFieldsExtractorTest.php

```diff
@@ -0,0 +1,141 @@
+        }
+    }
+
+    public function testExtractWithContentTypeNames(): void
+    {
+        $expression = '*/user/{first_name,last_name}';
+
+        $extractedFieldIds = $this->contentTypeFieldsExtractor->extractFieldsFromExpression($expression);
+
+        $firstNameField = $this->contentTypeHandler->getFieldDefinition($extractedFieldIds[0], 0);
+        $lastNameField = $this->contentTypeHandler->getFieldDefinition($extractedFieldIds[1], 0);
+
+        self::assertSame('first_name', $firstNameField->identifier);
+        self::assertSame('last_name', $lastNameField->identifier);
+    }
+
+    public function testExtractWithContentTypeAndGroupNames(): void
+    {
+        $expression = 'Users/user/{first_name,last_name}';
+
+        $extractedFieldIds = $this->contentTypeFieldsExtractor->extractFieldsFromExpression($expression);
+
+        $firstNameField = $this->contentTypeHandler->getFieldDefinition($extractedFieldIds[0], 0);
+        $lastNameField = $this->contentTypeHandler->getFieldDefinition($extractedFieldIds[1], 0);
+
+        self::assertSame('first_name', $firstNameField->identifier);
+        self::assertSame('last_name', $lastNameField->identifier);
+    }
+
+    public function testExtractWithContentTypeAndGroupNamesFailsWithTypesOutsideGroups(): void
+    {
+        self::expectException(LogicException::class);
+
+        $expression = 'Content/user/{first_name,last_name}';
+
+        $this->contentTypeFieldsExtractor->extractFieldsFromExpression($expression);
+    }
+
+    /**
+     * @dataProvider dataProviderForTestFieldIdWithinExpression
+     */
+    public function testFieldIdWithinExpression(string $expression): void
+    {
+        $contentType = $this->contentTypeService->loadContentTypeByIdentifier('folder');
+        $fieldDefinitions = $contentType->getFieldDefinitions();
+        $nameFieldDefinition = $fieldDefinitions->filter(
+            static fn (FieldDefinition $fieldDefinition): bool => $fieldDefinition->getIdentifier() === 'name'
+        )->first();
+
+        $result = $this->contentTypeFieldsExtractor->isFieldWithinExpression($nameFieldDefinition->getId(), $expression);
+
+        self::assertTrue($result);
+    }
+
+    /**
+     * @dataProvider dataProviderForTestFieldIdNotWithinExpression
+     */
+    public function testFieldIdNotWithinExpression(string $expression): void
+    {
+        $contentType = $this->contentTypeService->loadContentTypeByIdentifier('folder');
+        $fieldDefinitions = $contentType->getFieldDefinitions();
+        $nameFieldDefinition = $fieldDefinitions->filter(
+            static fn (FieldDefinition $fieldDefinition): bool => $fieldDefinition->getIdentifier() === 'name'
+        )->first();
+
+        $result = $this->contentTypeFieldsExtractor->isFieldWithinExpression($nameFieldDefinition->getId(), $expression);
+
+        self::assertFalse($result);
+    }
+
+    /**
+     * @return iterable<list<string>>
```

Comment:
You are actually returning an array with a **single** string in it, not a list.
```suggestion
     * @return iterable<array{string}>
```